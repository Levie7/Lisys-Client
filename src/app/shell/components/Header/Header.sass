@import 'src/style/lisys/settings/settings.border'
@import 'src/style/lisys/settings/settings.color'

.Header
    border-bottom: $border-width-regular solid $gray-lighter

    &_Dropdown
        .ant-dropdown-link
            padding-right: 15px
            position: relative

            &:after
                color: $gray
                position: absolute
                top: 50%
                right: 0
                margin-top: -2px
                display: inline-block
                width: 0
                height: 0
                margin-left: .255em
                vertical-align: .255em
                content: ''
                border-top: .3em solid
                border-right: .3em solid transparent
                border-bottom: 0
                border-left: .3em solid transparent

            &:hover
                &:after
                    color: $gray-light

    &_Avatar
        background: $gray-lighter !important
