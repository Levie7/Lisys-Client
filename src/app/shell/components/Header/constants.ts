export const headerAction = {
    logout: { en: 'Logout', id: 'Keluar' },
};
export const headerField = {
    role: { en: 'Role', id: 'Peran' },
    theme: { en: 'Theme Menu', id: 'Tema Menu' },
};
