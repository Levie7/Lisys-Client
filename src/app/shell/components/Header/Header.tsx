import { Avatar, Dropdown, Menu } from 'antd';
import * as React from 'react';

import { Lang } from 'src/core/api';

import { Icon } from 'src/shared/components/Icon';
import { queryForm } from 'src/shared/graphql';
import { getUserByUsername } from 'src/shared/graphql/User/schema.gql';
import { headerAction, headerField } from './constants';
import { Switch } from 'src/shared/components/Switch';

import { MenuTheme } from '../MainMenu/Menu';

require('./Header.sass');

interface HeaderProps extends Lang {
    auth: string | null;
    theme?: MenuTheme;

    handleTheme: (value: boolean) => void;
    logout: () => void;
}

export const Header = React.memo<HeaderProps>(({ auth, handleTheme, lang, logout, theme }) => {
    let queryUser = queryForm({
        query: getUserByUsername,
        variables: { username: auth },
    });
    if (queryUser.loading) {
        return null;
    }

    let user = queryUser.data?.getUserByUsername;

    let menu = (
        <Menu selectable={false}>
            <Menu.Item>
                <div className='rfq__widget__system-status__item'>
                    <strong>Hello, {user.name}</strong>
                    <div>
                        <strong>{headerField.role[lang]} :</strong> {user.role.name}
                    </div>
                </div>
            </Menu.Item>
            <Menu.Item>
                <div className='rfq__widget__system-status__item'>
                    <strong>{headerField.theme[lang]} : </strong>
                    <div>
                        <Switch
                            checkedChildren='Dark'
                            defaultChecked={theme === 'dark'}
                            handleChange={(value) => handleTheme(value)}
                            unCheckedChildren='Light'
                        />
                    </div>
                </div>
            </Menu.Item>
            <Menu.Item>
                <div onClick={handleLogout}>
                    <i className='topbar__dropdownMenuIcon icmn-exit' /> {headerAction.logout[lang]}
                </div>
            </Menu.Item>
        </Menu>
    );

    function handleLogout() {
        logout();
    }

    return (
        <div className='Header d-flex fj-between px-4'>
            <div />
            <div className='Header_Dropdown'>
                <Dropdown overlay={menu} trigger={['click']} placement='bottomRight'>
                    <a className='ant-dropdown-link' href='/'>
                        <Avatar
                            className='Header_Avatar'
                            shape='square'
                            size='large'
                            icon={Icon.user}
                        />
                    </a>
                </Dropdown>
            </div>
        </div>
    );
});
