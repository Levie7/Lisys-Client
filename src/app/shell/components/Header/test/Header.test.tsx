import { shallow } from 'enzyme';
import React from 'react';

import { Header } from '../Header';

describe('Header', () => {
    const wrap = shallow(<Header />);

    it('<Header/> - should render header component correctly', () => {
        expect(wrap).toMatchSnapshot();
    });

    it('should render header', () => {
        expect(wrap.find('.Header').exists()).toBeTruthy();
    });
});
