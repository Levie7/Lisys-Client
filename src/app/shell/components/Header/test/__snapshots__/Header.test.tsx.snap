// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header <Header/> - should render header component correctly 1`] = `ShallowWrapper {}`;
